### PR TITLE
Feature/il 4580 change powerUp curve 

### PR DESCRIPTION
**Pull request by pete-ipor**

### CODE REVIEW

File: contracts/libraries/math/MiningCalculationAccountPowerUp.sol [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    uint256 constant SLOPE_3 = 15e17; //   1.5
+    uint256 constant BASE_3 = 28e16; //    0.28
+
+    uint256 constant SLOPE_4 = 1; //   1.0
+    uint256 constant BASE_4 = 31e16; //    0.31
+
+    uint256 constant SLOPE_5 = 5e17; //   0.5
+    uint256 constant BASE_5 = 35e16; //    0.35
+
+    using SafeCast for uint256;
```
Comment (mario-ipor) [inline contracts/libraries/math/MiningCalculationAccountPowerUp.sol:39]: using on top

File: contracts/libraries/math/MiningCalculationAccountPowerUp.sol [hunk 2]
```diff
@@ -0,0 +1,136 @@
+// SPDX-License-Identifier: BSD-3-Clause
+pragma solidity 0.8.20;
+
+import "abdk-libraries-solidity/ABDKMathQuad.sol";
+import "@openzeppelin/contracts/utils/math/SafeCast.sol";
+
+import "../errors/Errors.sol";
+import "./MathOperation.sol";
+
+struct AccountPowerUpData {
```
Comment (mario-ipor) [inline contracts/libraries/math/MiningCalculationAccountPowerUp.sol:10]: Put here information about decimals for every parameter.

File: contracts/mining/CalculateWeightedLpTokenBalanceArbitrum.sol [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        address wstEthStEthExchangeRateOracle_
+    ) public view returns (uint256) {
+        if (lpToken_ != lpwstEth_) {
+            return lpTokenBalance_;
+        }
+
+        // @dev returned value has 8 decimal, on arbitrum 0x639Fe6ab55C921f74e7fac1ee960C0B6293ba612
+        (, int256 answerEthUsd, , , ) = AggregatorV3Interface(ethUsdOracle_).latestRoundData();
+        // @dev returned value has 18 decimal, on arbitrum 0xe59eba0d492ca53c6f46015eea00517f2707dc77, not avable on mainnet
+        (, int256 answerWstEthStEth, , , ) = AggregatorV3Interface(wstEthStEthExchangeRateOracle_)
```
Comment (mario-ipor) [inline contracts/mining/CalculateWeightedLpTokenBalanceArbitrum.sol:26]: comment "not avaible on Ethereum Mainnet"

File: contracts/mining/LiquidityMiningArbitrum.sol [hunk 4]
```diff
@@ -11,35 +12,35 @@ contract LiquidityMiningArbitrum is LiquidityMining {
 
     address internal immutable ethUsdOracle;
     address internal immutable wstEthStEthExchangeRateOracle;
+    address internal immutable lpwstEth;
 
     constructor(
         address routerAddress,
         address ethUsdOracleInput,
```
Comment (mario-ipor) [inline contracts/mining/LiquidityMiningArbitrum.sol:19]: _ instead Input

File: contracts/libraries/math/MiningCalculationAccountPowerUp.sol [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+                data_.horizontalShift
+            );
+
+            result = ABDKMathQuad.add(
+                data_.verticalShift,
+                _log(_toQuadruplePrecision(data_.logBase, 1e18), underLog)
+            );
+            bytes16 resultD18 = ABDKMathQuad.mul(result, ABDKMathQuad.fromUInt(1e18));
+
+            //The number 222392421336447926 is the value by which we want to lower the default function values. This value can never be negative.
```
Comment (mario-ipor) [inline contracts/libraries/math/MiningCalculationAccountPowerUp.sol:75]: add /// @dev

File: contracts/interfaces/types/LiquidityMiningTypes.sol [hunk 6]
```diff
@@ -81,4 +81,13 @@ library LiquidityMiningTypes {
         address lpToken;
         LiquidityMiningTypes.GlobalRewardsIndicators indicators;
     }
+
+    struct PoolPowerUpModifier {
+        // @dev 10 decimals
```
Comment (pete-ipor) [inline contracts/interfaces/types/LiquidityMiningTypes.sol:86]: ///

**Review by mario-ipor — APPROVED** [review]

**Pull request merged**